### PR TITLE
[Admin] component generator fixes and specs

### PR DESCRIPTION
**Pull request by elia**
## Summary

There were a couple of bugs in the generator. A basic spec was added to avoid future regressions.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: admin/spec/generator/solidus_admin/component_generator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+require 'rails/generators'
+require 'rails/generators/app_base'
+require "rails/generators/testing/behaviour"
+require "rails/generators/testing/setup_and_teardown"
+require "rails/generators/testing/assertions"
+require 'generators/solidus_admin/component/component_generator'
+
+RSpec.describe SolidusAdmin::ComponentGenerator, type: :generator do
+  include Rails::Generators::Testing::Behaviour
+  include Rails::Generators::Testing::SetupAndTeardown
```
Comment (rainerdema) [inline admin/spec/generator/solidus_admin/component_generator_spec.rb:14]: Looking at [railties/lib/rails/generators/testing/setup_and_teardown.rb](https://github.com/rails/rails/blob/34bd524a9d7580ceecbf4a6dd9dced6296e20f97/railties/lib/rails/generators/testing/setup_and_teardown.rb) it appears that this `include` may not be necessary.

File: admin/spec/generator/solidus_admin/component_generator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'rails/version'
+require 'rails/generators'
+require 'rails/generators/app_base'
+require "rails/generators/testing/behaviour"
+require "rails/generators/testing/setup_and_teardown"
+require "rails/generators/testing/assertions"
```
Comment (rainerdema) [inline admin/spec/generator/solidus_admin/component_generator_spec.rb:9]: `rails/generators/testing/assertions` appears to be unused.

**Review by rainerdema — APPROVED** [review]
Just a couple of comments but the fix looks great! 👏

File: admin/spec/generator/solidus_admin/component_generator_spec.rb (hunk 1 above)
Comment (elia) [reply admin/spec/generator/solidus_admin/component_generator_spec.rb]: Thanks fixed, I also moved the generic includes and requires to the spec_helper.rb

**Review by waiting-for-dev — APPROVED** [review]
Thanks!

**Pull request merged**